### PR TITLE
Add the call to the plugins installation in the "move exported files" process

### PR DESCRIPTION
**Pull request by garnier-quentin**
## Description

It's a part of the new feature. 

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [X] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

It needs the update of gorgone and ppm.

## Checklist

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 1]
```diff
@@ -85,6 +85,21 @@ public function __construct($centreon_path, \Pimple\Container $dependencyInjecto
...
+     */
+    private function isSyncPlugins()
+    {
+        $DBRESULT = $this->DB->query("SELECT `value` FROM `options` WHERE `key` = 'pack_sync_plugins'");
+        if ($data = $DBRESULT->fetchRow()) {
+            return ($data['value'] == 1 ? 1 : 0);
+        }
+
+        return 0;
+    }
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:101]:
```suggestion
    /**
     * Get the option pack_sync_plugins
     *
     * @return bool
     */
    private function isSyncPlugins(): bool
    {
        $cbResult = $this->DB->query("SELECT `value` FROM `options` WHERE `key` = 'pack_sync_plugins'");
        if ($data = $cbResult->fetch()) {
            return $data['value'] === '1';
        }

        return false;
    }
```

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 2]
```diff
@@ -181,7 +196,8 @@ public function pollerReload($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RELOAD:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec("echo '" . ($syncPlugins == 1 ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:200]:
```suggestion
        exec("echo '" . ($syncPlugins ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 3]
```diff
@@ -249,7 +265,8 @@ public function pollerRestart($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RESTART:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec("echo '" . ($syncPlugins == 1 ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:269]:
```suggestion
        exec("echo '" . ($syncPlugins ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```

File: www/class/config-generate/plugins.class.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ */
+
+class Plugins extends AbstractObjectJSON
+{
+    protected $generate_filename = 'plugins.json';
+    protected $pluginsCache = null;
+    protected $plugins = [];
+
+    private function getPlugins()
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:43]: missing phpdoc

File: www/class/config-generate/plugins.class.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            SELECT
+                command, package_name, MAX(version) as version
+            FROM mod_ppm_pluginpack_command_version
+            GROUP BY command
+        ");
+        $stmt->execute();
+        $this->pluginsCache = $stmt->fetchAll(PDO::FETCH_GROUP | PDO::FETCH_UNIQUE | PDO::FETCH_ASSOC);
+    }
+
+    public function addCommand($commandLine)
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:55]: missing phpdoc

File: www/class/config-generate/plugins.class.php [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+        $command = $matches[1];
+        if (!isset($this->pluginsCache[$command])) {
+            return;
+        }
+
+        $this->plugins[ $this->pluginsCache[$command]['package_name'] ] = $this->pluginsCache[$command]['version'];
+    }
+
+    public function generate()
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:73]: missing phpdoc

File: www/class/config-generate/plugins.class.php [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $this->content = json_encode($this->plugins, JSON_UNESCAPED_SLASHES|JSON_PRETTY_PRINT);
+        $this->writeFile($this->backend_instance->getPath());
+    }
+
+    public function reset()
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:79]: missing phpdoc

File: www/class/config-generate/plugins.class.php (hunk 5 above)
Comment (kduret) [inline www/class/config-generate/plugins.class.php:55]:
```suggestion
    public function addCommand(string $commandLine)
```

File: www/include/configuration/configGenerate/xml/restartPollers.php [hunk 8]
```diff
@@ -204,6 +204,8 @@
         }
     }
 
+    $syncPlugins = isSyncPlugins();
```
Comment (kduret) [inline www/include/configuration/configGenerate/xml/restartPollers.php:207]:
```suggestion
    $syncPlugins = isSyncPlugins($pearDB);
```

File: www/include/configuration/configGenerate/xml/restartPollers.php [hunk 9]
```diff
@@ -213,7 +215,7 @@
     foreach ($poller as $host) {
         if ($ret["restart_mode"] == 1) {
             if ($fh = @fopen($centcore_pipe, 'a+')) {
-                fwrite($fh, "RELOAD:" . $host["id"] . "\n");
+                fwrite($fh, ($syncPlugins == 1 ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"] . "\n");
```
Comment (kduret) [inline www/include/configuration/configGenerate/xml/restartPollers.php:218]:
```suggestion
                fwrite($fh, ($syncPlugins ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"] . "\n");
```

File: www/include/configuration/configGenerate/xml/restartPollers.php [hunk 10]
```diff
@@ -229,7 +231,7 @@
             );
         } elseif ($ret["restart_mode"] == 2) {
             if ($fh = @fopen($centcore_pipe, 'a+')) {
-                fwrite($fh, "RESTART:" . $host["id"] . "\n");
+                fwrite($fh, ($syncPlugins == 1 ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"] . "\n");
```
Comment (kduret) [inline www/include/configuration/configGenerate/xml/restartPollers.php:234]:
```suggestion
                fwrite($fh, ($syncPlugins ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"] . "\n");
```

File: www/include/configuration/configGenerate/DB-Func.php [hunk 11]
```diff
@@ -53,3 +53,20 @@ function getCentreonBrokerDirCfg($ns_id)
...
+{
+    global $pearDB;
+
+    $res = $pearDB->query("SELECT `value` FROM `options` WHERE `key` = 'pack_sync_plugins'");
+    if ($data = $res->fetch()) {
+        return ($data['value'] == 1 ? 1 : 0);
+    }
+
+    return 0;
+}
```
Comment (kduret) [inline www/include/configuration/configGenerate/DB-Func.php:72]:
```suggestion
/**
 * Get the option pack_sync_plugins
 *
 * @param CentreonDB $pearDB
 * @return bool
 */
function isSyncPlugins($pearDB): bool
{
    $res = $pearDB->query("SELECT `value` FROM `options` WHERE `key` = 'pack_sync_plugins'");
    if ($data = $res->fetch()) {
        return $data['value'] === '1';
    }

    return false;
}
```

**Review by callapa — COMMENTED** [review]
> It's a part of the new feature.

What features? No description?

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 12]
```diff
@@ -181,7 +196,8 @@ public function pollerReload($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RELOAD:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec("echo '" . ($syncPlugins ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:200]:
```suggestion
        exec(
            "echo '" . ($syncPlugins ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"]
            . "' >> " . $this->centcore_pipe, $stdout, $return_code
        );
```

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 13]
```diff
@@ -249,7 +265,8 @@ public function pollerRestart($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RESTART:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec("echo '" . ($syncPlugins ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:269]:
```suggestion
        exec(
            "echo '" . ($syncPlugins ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"]
            . "' >> " . $this->centcore_pipe, $stdout, $return_code
        );
```

File: www/class/config-generate/command.class.php [hunk 14]
```diff
@@ -105,6 +105,8 @@ public function generateFromCommandId($command_id)
         $command_line = str_replace("\n", " \\\n", $command_line);
         $command_line = str_replace("\r", "", $command_line);
 
+        Plugins::getInstance($this->dependencyInjector)->addCommand($command_line);
+
         if (!is_null($this->commands[$command_id]['enable_shell']) &&
```
Comment (kduret) [inline www/class/config-generate/command.class.php:110]:
```suggestion
        if (
            !is_null($this->commands[$command_id]['enable_shell']) &&
```

File: www/class/config-generate/plugins.class.php (hunk 4 above)
Comment (kduret) [inline www/class/config-generate/plugins.class.php:43]:
```suggestion
    /**
     * store plugin packs and their version in cache
     */
    private function getPlugins()
```

File: www/class/config-generate/plugins.class.php [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            SELECT
+                command, package_name, MAX(version) as version
+            FROM mod_ppm_pluginpack_command_version
+            GROUP BY command
+        ");
+        $stmt->execute();
+        $this->pluginsCache = $stmt->fetchAll(PDO::FETCH_GROUP | PDO::FETCH_UNIQUE | PDO::FETCH_ASSOC);
+    }
+
+    public function addCommand(string $commandLine)
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:55]:
```suggestion
    /**
     * add command to generate
     */
    public function addCommand(string $commandLine)
```

File: www/class/config-generate/plugins.class.php [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+        $command = $matches[1];
+        if (!isset($this->pluginsCache[$command])) {
+            return;
+        }
+
+        $this->plugins[ $this->pluginsCache[$command]['package_name'] ] = $this->pluginsCache[$command]['version'];
+    }
+
+    public function generate()
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:73]:
```suggestion
    /**
     * generate file
     */
    public function generate()
```

File: www/class/config-generate/plugins.class.php [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        $this->content = json_encode($this->plugins, JSON_UNESCAPED_SLASHES|JSON_PRETTY_PRINT);
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:75]:
```suggestion
        $this->content = json_encode($this->plugins, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT);
```

File: www/class/config-generate/plugins.class.php [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+        $this->writeFile($this->backend_instance->getPath());
+    }
+
+    public function reset()
```
Comment (kduret) [inline www/class/config-generate/plugins.class.php:79]:
```suggestion
    /**
     * reset cache
     */
    public function reset()
```

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 19]
```diff
@@ -181,7 +196,11 @@ public function pollerReload($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RELOAD:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec(
+            "echo '" . ($syncPlugins ? 'ENGINERELOAD' : 'RELOAD') . ":" . $host["id"]
+            . "' >> " . $this->centcore_pipe, $stdout, $return_code
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:202]:
```suggestion
                . "' >> " . $this->centcore_pipe,
            $stdout,
            $return_code
```

File: www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 20]
```diff
@@ -249,7 +268,11 @@ public function pollerRestart($variables)
         $host = $result->fetch();
         $result->closeCursor();
 
-        exec("echo 'RESTART:" . $host["id"] . "' >> " . $this->centcore_pipe, $stdout, $return_code);
+        $syncPlugins = $this->isSyncPlugins();
+        exec(
+            "echo '" . ($syncPlugins ? 'ENGINERESTART' : 'RESTART') . ":" . $host["id"]
+            . "' >> " . $this->centcore_pipe, $stdout, $return_code
```
Comment (kduret) [inline www/class/centreon-clapi/centreon.Config.Poller.class.php:274]:
```suggestion
                . "' >> " . $this->centcore_pipe,
            $stdout,
            $return_code
```

**Pull request closed without merging**